kit/inputs/src/UIAmountInput/UIAmountInputContent.tsx

```diff
@@ -0,0 +1,234 @@
+import * as React from 'react';
+import Animated, { runOnUI, useAnimatedRef, useSharedValue } from 'react-native-reanimated';
+import { NativeModules, Platform, TextStyle, View } from 'react-native';
+import { makeStyles, Theme, ColorVariants, useTheme, UILabelAnimated } from '@tonlabs/uikit.themes';
+import { UILayoutConstant } from '@tonlabs/uikit.layout';
+import type { BigNumber } from 'bignumber.js';
+import { uiLocalized } from '@tonlabs/localization';
+import {
+    UIAmountInputRef,
+    UIAmountInputProps,
+    UIAmountInputMessageType,
+    UIAmountInputDecimalAspect,
+} from './types';
+import { AmountInputContext } from './constants';
+import {
+    useAmountInputHandlers,
+    useAmountInputHover,
+    useExpandingValue,
+    useExtendedRef,
+    useDefaultValue,
+    useFormatText,
+    useSetText,
+    useInputVerticalMovementStyle,
+    usePlaceholderAttributes,
+    useInputMessageType,
+} from './hooks';
+import { UITextView, UITextViewRef } from '../UITextView';
+import { TapHandler } from './TapHandler';
+import { InputMessage } from '../InputMessage';
+import { FloatingLabel } from './FloatingLabel';
+import { useInputChildren } from '../useInputChildren';
+
+const UITextViewAnimated = Animated.createAnimatedComponent(UITextView);
+
+NativeModules.UIKitInputModule?.install();
+
+const decimalSeparator = uiLocalized.localeInfo.numbers.decimal;
+
+export const UIAmountInputContent = React.forwardRef<UIAmountInputRef, UIAmountInputProps>(
+    function UIAmountInputContent(
+        { children, placeholder, defaultAmount, ...props }: UIAmountInputProps,
+        forwardedRef: React.Ref<UIAmountInputRef>,
+    ) {
+        const {
+            editable = true,
+            onFocus,
+            onBlur,
+            onHover,
+            onSelectionChange,
+            onChangeAmount: onChangeAmountProp,
+            decimalAspect = UIAmountInputDecimalAspect.Precision,
+            multiline,
+            message,
+            messageType = UIAmountInputMessageType.Info,
+            label,
+        } = props;
+        // @ts-ignore
```

Comment:
What's the problem?

Reply:
The types don't match with `useAnimatedRef`. For this purposes it doesn't matter.